app/src/main/java/com/example/datarsd1/mdbsocials_2/AddSocial.java

```diff
@@ -0,0 +1,124 @@
+                if (eventimageuri==null|| newTitleText.getText()==null || newDateText.getText()==null || newDescText.getText()==null ||
+                        newTitleText.getText().toString().length()==0 || newDateText.getText().toString().length()==0 || newDescText.getText().toString().length()==0)
+                {
+                    Toast.makeText(getApplicationContext(),"Please Enter All Fields Properly",Toast.LENGTH_LONG).show();
+                }else
+                {
+                    final DatabaseReference ref = FirebaseDatabase.getInstance().getReference();
+                    final String key = ref.child("Socials").push().getKey();
+                    StorageReference storageRef = FirebaseStorage.getInstance().getReferenceFromUrl("gs://mdbsocials-2.appspot.com");
+                    StorageReference riversRef = storageRef.child("eventPics/" + key + ".jpg");
+
+                    String name = newTitleText.getText().toString();
+                    String date = newDateText.getText().toString();
+                    String desc = newDescText.getText().toString();
+                    String host = mAuth.getCurrentUser().getEmail();
+
+                    FirebaseUtils.writeEventToDB(mDatabase, riversRef, eventimageuri, key, name, date, desc, host, mAuth);
+                    startActivity(new Intent(getApplicationContext(),UserFeedActivity.class));
+                }
+                break;
+            case R.id.newEventImage:
+                AlertDialog alertDialog = new AlertDialog.Builder(AddSocial.this).create();
+                alertDialog.setTitle("Image Upload Options");
+                alertDialog.setButton(AlertDialog.BUTTON_NEUTRAL, "Take a Photo", new DialogInterface.OnClickListener()
+                {
+                    public void onClick(DialogInterface dialog, int which)
+                    {
+                        dialog.dismiss();
+                        Intent takePictureIntent = new Intent(android.provider.MediaStore.ACTION_IMAGE_CAPTURE);
+
+                        ContentValues values = new ContentValues();
+                        values.put(MediaStore.Images.Media.TITLE, "Image");
+                        Uri tempuri = getContentResolver().insert(MediaStore.Images.Media.EXTERNAL_CONTENT_URI, values);
+                        takePictureIntent.putExtra(MediaStore.EXTRA_OUTPUT, tempuri);
+                        startActivityForResult(takePictureIntent, 0);
+                    }
+                });
+                alertDialog.setButton(AlertDialog.BUTTON_POSITIVE, "Upload from Gallery", new DialogInterface.OnClickListener()
+                {
+                    public void onClick(DialogInterface dialog, int which) {
+                        dialog.dismiss();
+                        Intent intent = new Intent(Intent.ACTION_GET_CONTENT);
+                        intent.setType("image/*");
+                        startActivityForResult(intent, 1);                            }

```

Comment:
Instead of doing startActivityForResult(intent, 1), put 1 in a public static final variable. For example, you could do:
public static final int CHOOSE_PHOTO_REQUEST = 1. That way, in onActivityResult in the bottom, when you check whether requestCode == 1, you know what it is actually checking for (instead you would check requestCode == CHOOSE_PHOTO_REQUEST).